backend/server.go

```diff
@@ -8,29 +8,97 @@ import (
 	"github.com/go-chi/chi/v5"
 	"github.com/go-chi/chi/v5/middleware"
 	"github.com/go-chi/cors"
+	"golang.org/x/crypto/acme/autocert"
 	"net/http"
+	"os"
+	"path/filepath"
 	"strings"
 )
 
 // See https://github.com/go-chi/chi for documentation
 
-const PORT = 5001 // todo: put into environment var
+func init() {
+
+	if value, ok := os.LookupEnv("NODE_ENV"); ok && value == "production" {
+		allVars := []envVar{
+			{key: "API_BASE_URL", preset: "/api"},
+		}
+
+		err := filepath.Walk("/static/_next", func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if !info.IsDir() && filepath.Ext(path) == ".js" {
+				err := replaceVarsInFile(path, allVars)
+				if err != nil {
+					fmt.Printf("Error replacing vars in file %s: %v\n", path, err)
+				}
+			}
+
+			return nil
+		})
+
+		if err != nil {
+			fmt.Printf("Error walking the path: %v\n", err)
+		}
+	}
+}
 
 func main() {
+	var origins []string
+	if value, ok := os.LookupEnv("NODE_ENV"); ok && value == "production" {
+		if domain, ok := os.LookupEnv("WEBAUTHN_DOMAIN"); ok {
+			origins = []string{fmt.Sprintf("https://%s*", domain)}
+		} else {
+			origins = []string{"https://app.lynx-locks.com*"}
+		}
+	} else {
+		origins = []string{"http://localhost:3000*"}
+	}
+
 	r := chi.NewRouter()
 	r.Use(middleware.Logger)
 	r.Use(cors.Handler(cors.Options{
-		AllowedOrigins: []string{"http://localhost:3000*"},
+		AllowedOrigins: origins,
 		AllowedMethods: []string{"GET", "POST", "PUT", "DELETE", "OPTIONS"},
 		AllowedHeaders: []string{"Accept", "Authorization", "Content-Type", "X-CSRF-Token"},
 	}))
 	config.Connect()
 	FileServer(r, "/", http.Dir("./static"))
 	r.Mount("/api", api())
 
-	fmt.Printf("Server running on port %d\n", PORT)
-	err := http.ListenAndServe(fmt.Sprintf(":%d", PORT), r)
-	helpers.CheckErr(err)
+	var PORT int
+	if value, ok := os.LookupEnv("NODE_ENV"); ok && value == "production" {
+		PORT = 443
+
+		m := &autocert.Manager{
+			Cache:      autocert.DirCache("cert-cache"),
+			Prompt:     autocert.AcceptTOS,
+			HostPolicy: autocert.HostWhitelist("app.lynx-locks.com"),
+		}
+
+		server := &http.Server{
+			Addr:      fmt.Sprintf(":%d", PORT),
+			Handler:   r,
+			TLSConfig: m.TLSConfig(),
+		}
+
+		fmt.Printf("Server running on port %d\n", PORT)
+		go func() {
+			err := http.ListenAndServe(":80", m.HTTPHandler(nil))
```

Comment:
Why do we have port 80 and port 443 here?

Reply:
This is for communication for the http-01 challenge for certs. I'll add a comment.